include/motor-control/simulation/brushed_motor_interrupt_driver.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "common/core/freertos_message_queue.hpp"
+#include "common/core/freertos_task.hpp"
+#include "common/core/logging.h"
+#include "motor-control/core/brushed_motor/brushed_motor_interrupt_handler.hpp"
+#include "motor-control/core/motor_messages.hpp"
+#include "motor-control/simulation/sim_motor_hardware_iface.hpp"
+
+namespace brushed_motor_interrupt_driver {
+template <class StatusClient, class BrushedMotorHardware>
+class BrushedMotorInterruptDriver {
+    using InterruptQueue = freertos_message_queue::FreeRTOSMessageQueue<
+        motor_messages::BrushedMove>;
+    using InterruptHandler =
+        brushed_motor_handler::BrushedMotorInterruptHandler<
+            freertos_message_queue::FreeRTOSMessageQueue, StatusClient>;
+
+  public:
+    BrushedMotorInterruptDriver(InterruptQueue& q, InterruptHandler& h,
+                                BrushedMotorHardware& iface)
+        : task_entry{q, h, iface}, task(task_entry) {
+        task.start(5, "sim_motor_isr");
+    }
+
+  private:
+    struct TaskEntry {
+        TaskEntry(InterruptQueue& q, InterruptHandler& h,
+                  BrushedMotorHardware& motor_iface)
+            : queue{q}, handler{h}, iface{motor_iface} {}
+
+        void operator()() {
+            while (true) {
+                auto move = motor_messages::BrushedMove{};
+                if (queue.peek(&move, queue.max_delay)) {
+                    LOG("Enabling motor interrupt handler for group %d, seq "
+                        "%d, duration %ld",
+                        move.group_id, move.seq_id, move.duration);
+                    do {
+                        if (queue.try_read(&move, 0)) {
```

Comment:
Similarly to the stepper simulator, I don't think we want to read the messages here. `brushed_motor_interrupt_handler` already reads from the queue using `try_read_isr`. We just want to peek here to see if the next move should trigger a stop condition because, while we want to make sure the interrupt handler sees the appropriate stop condition, we still need the handler to see the new move and set it as the active buffered movement.